plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -1217,3 +1217,275 @@ SELECT
 	,DATABASEPROPERTYEX(DB_NAME(), 'Updateability') as replica_updateability
 FROM sys.dm_os_schedulers AS s
 `
+
+const sqlAzureMIPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+IF SERVERPROPERTY('EngineEdition') <> 8 BEGIN /*not Azure Managed Instance*/
+	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - Connection string Server:'+ @@SERVERNAME + ',Database:' + DB_NAME() +' is not an Azure Managed Instance. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END;
+
```

Comment:
One general comment: do we have support for only collecting QDS data from some databases on an MI instance? For instances with many (dozens) of databases, it is likely that customers may want to collect only from a few. We may also find that collecting from all is too heavy. Can we add configurations along the lines of IncludeQueryStoreForDatabases and/or ExcludeQueryStoreForDatabases? Default would be collect from all.

Reply:
@dimitri-furman, currently telegraf doesn't support any DB filtering. If we add simple list IncludeQueryStoreForDatabases to a sqlserver plugin config section it will affect all servers in this section and in many cases this behavior may be undesirable. In order to allow per-server DB filtering the sqlserver plugin configuration schema and the plugin's config processing may require significant changes. Also DB filteringng may be also applied to other queries. That is why I think that DB filtering should be implemented as a separate PR.

Reply:
But that would disable all collection, right? The intent is to disable voluminous and heavy QDS collection for databases where it's not needed.

Reply:
@dimitri-furman, why would it disable all collection? A custemer can simply give access to DBs he wants to monitor. Other sql queries (not QDS) don't require per-DB access, they use instance-level views and won't be affected. Also a customer can simply create a separate account for QDS monitoring.
In any case limiting access now is the only way to select DBs for QDS monitoring. In one of my previous comments I wrote why adding additional configuration parameter in this PR is a bad idea.